pkg/apis/serving/v1beta1/component.go

```diff
@@ -197,6 +203,18 @@ type ExternalMetricSource struct {
 	Target MetricTarget `json:"target,omitempty"`
 }
 
+// PodsMetricSource indicates how to scale on a metric describing each pod in
+// the current scale target (for example, transactions-processed-per-second).
+// The values will be averaged together before being compared to the target
+// value.
+type PodsMetricSource struct {
+	// metric identifies the target metric by name and selector
+	Metric MetricSource `json:"metric,omitempty"`
```

Comment:
I think we might want a separate MetricSource type for PodMetrics, the backend for PodMetrics is opentelemetry while for ExtenralMetrics it is prometheus/graphite.